### PR TITLE
Update Intel architectures

### PR DESCRIPTION
**Pull request by masterleinad**

### CODE REVIEW

File: docs/source/keywords.rst [hunk 1]
```diff
@@ -306,8 +306,32 @@ Architecture Keywords
       * Optimize for HSW architecture
       * ``OFF``
 
-    * * ``Kokkos_ARCH_INTEL_GEN``
-      * Optimize for Intel GPUs Gen9+
+    * * ``KOKKOS_ARCH_INTEL_GEN``
+      * Optimize for Intel GPUs, JIT
```
Comment (dalg24) [inline docs/source/keywords.rst:310]: Do we explain somewhere what JIT means and how it differs from other options?

File: docs/source/keywords.rst [hunk 2]
```diff
@@ -378,6 +402,10 @@ Architecture Keywords
       * Optimize for SNB architecture
       * ``OFF``
 
+    * * ``Kokkos_ARCH_SPR``
+      * Optimize for Sapphire Rapids architecture
```
Comment (dalg24) [inline docs/source/keywords.rst:406]: Say Intel

Comment (dalg24) [reply docs/source/keywords.rst:406]: Nevermind I see we don't say anything for other Intel CPU architecture.  We can improve later.

File: docs/source/keywords.rst (hunk 1 above)
Comment (masterleinad) [reply docs/source/keywords.rst:310]: No, as far as I know, this is the only occurrence of something JIT related in `Kokkos`. Do you think it's sufficient to just spell it out here?

Comment (dalg24) [reply docs/source/keywords.rst]: You will need to write a paragraph that explains the alternatives ahead-of-time VS just-in-time and what these entail.
Adding as a follow-up is fine if you prefer

Comment (masterleinad) [reply docs/source/keywords.rst]: Fair enough. I gave it a shot. 🙂

**Review by dalg24 — APPROVED** [review]

**Review by Rombur — APPROVED** [review]

File: docs/source/API/core/Macros.rst [hunk 3]
```diff
@@ -254,3 +256,8 @@ Architectures
 | ``KOKKOS_ARCH_VEGA90A``           | Optimize for AMD GPU MI200 series GFX90A (GPU)                                    |
 +-----------------------------------+-----------------------------------------------------------------------------------+
 
+*) ``KOKKOS_ARCH_INTEL_GEN`` enables Just-In-Time compilation for Intel GPUs whereas all the other flags for Intel compilers
+request Ahead-Of-Time compilation. Just-In-Time compilation means that the compiler is invoked again when the binaries created
+are actually executed and only at that point the architecture to compile for is determined. On the other hand, Ahead-Of-Time
+compilation describes the standard model where the compiler is only invoked once to create the binary and the architecture to
+compile for is determined before the program is run.
```
Comment (dalg24) [inline docs/source/API/core/Macros.rst:263]: I just realized you commented by the macros.  It is actually more useful by the CMake options

**Pull request merged**